### ISSUE TITLE
Finish Building Test Suite Framework for VIP Search

### ISSUE THREAD

**Issue by jakewrfoster** [alleyinteractive/elasticsearch-extensions#69]
### Description

Currently the test suite does not properly load and run VIP Search (and thus ElasticPress). Before we can set up tests against VIP Search, we will need to build the framework for VIP Search on ElasticPress. This will involve revisiting the bootstrapping and set up of VIP Search.

### Use Case

For future developers to be able to add tests quickly and easily.

**Comment by renatonascalves** [comment]
@jakewrfoster Can you outline, maybe with code or links, to some of the issues you run into?

I know ElasticPress has some abstract test classes that could help here, and I planned to use them when I set up the test suite for this plugin. But I haven't got into really adding tests per se, so I don't know exactly what are the issues are suggesting to finish here.

**Comment by jakewrfoster** [comment]
@renatonascalves currently, VIP Search is not being bootstrapped correctly. ElasticPress is not being initialized early enough for VIP Search's hooks to fire properly. VIP Search also create a custom table, which currently isn't being created, but needs to be in order to fully leverage VIP Search. The setup and teardown methods are not firing currently, either. These should be updated from `set_up_before_class` to `setUpBeforeClass`, for example.

There are a lot of little things that need to shored up for the sake of VIP Search.

**Comment by jakewrfoster** [comment]
I realized something was off when writing tests for the phrase matching feature and saw that certain ElasticPress filters were not firing. A version of ElasticPress was being initialized, but it was happening directly (IIRC) and the way that it needs to happen is via VIP Search/MU functions.

**Comment by jakewrfoster** [comment]
I have updated this issue to include only VIP Search tests. The SearchPress tests need to be updated in order for them to become functional as well. I will open a new issue for the SP tests.
